javers-persistence-sql/src/main/java/org/javers/repository/sql/schema/FixedSchemaFactory.java

```diff
@@ -111,9 +112,9 @@ private Schema commitPropertiesTableSchema(Dialect dialect) {
         RelationBuilder relationBuilder = schema.addRelation(tableName.nameWithSchema());
         relationBuilder
             .primaryKey(tableName.localName() + "_pk").using(COMMIT_PROPERTY_COMMIT_FK, COMMIT_PROPERTY_NAME).and()
-            .withAttribute().string(COMMIT_PROPERTY_NAME).withMaxLength(MAX_INDEX_KEY_LEN_IN_MYSQL).and()
+            .withAttribute().string(COMMIT_PROPERTY_NAME).withMaxLength(MAX_INDEX_KEY_LEN_IN_MYSQL).notNull().and()
             .withAttribute().string(COMMIT_PROPERTY_VALUE).withMaxLength(600).and();
-        foreignKey(tableName, COMMIT_PROPERTY_COMMIT_FK, getCommitTableNameWithSchema(), COMMIT_PK, relationBuilder);
+        foreignKey(tableName, COMMIT_PROPERTY_COMMIT_FK, true, getCommitTableNameWithSchema(), COMMIT_PK, relationBuilder);
```

Comment:
why you want this additional not null constraint and why have you added this only for this fk?
SNAPSHOT_GLOBAL_ID_FK and SNAPSHOT_COMMIT_FK are alse not null

Reply:
I mean we have not null constraints mainly for primary keys so why to add not nulls for other columns?

Reply:
I added that because COMMIT_PROPERTY_COMMIT_FK is part of a primary key. DB2 throws -542 error:

"-542   column-name CANNOT BE A COLUMN OF A HASH KEY, PRIMARY KEY, A UNIQUE CONSTRAINT, OR A PARENT KEY BECAUSE IT CAN CONTAIN NULL VALUES"

Reply:
ok, so `if Db2 ...` https://github.com/javers/javers/pull/624/commits/ce1f44a9e22b0aa04724b7a13e1ee7d38a5d5c99 ?

Reply:
fixed my typo, so final PR would be https://github.com/javers/javers/pull/624/files , right?

Reply:
Yes, you're right